### ISSUE TITLE
[CI] MixedClusterClientYamlTestSuiteIT search.inner_hits/10_basic/Inner hits with disabled _source fails

### ISSUE THREAD

**Issue by tlrx** [elastic/elasticsearch#67079]
This test failed multiple times in the last 24 hours after it was introduced in #66725.

**Build scans**:
https://gradle-enterprise.elastic.co/s/5lsjfiywoegee
https://gradle-enterprise.elastic.co/s/rroum4p4bt7gm
https://gradle-enterprise.elastic.co/s/fbvmbxii6izv4
https://gradle-enterprise.elastic.co/s/df5ckrz6lgv5i

**Repro line**:
```
./gradlew ':qa:mixed-cluster:v7.10.1#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.inner_hits/10_basic/Inner hits with disabled _source}" -Dtests.seed=14989D9D5778D988 -Dtests.security.manager=true -Dtests.locale=he-IL -Dtests.timezone=Arctic/Longyearbyen -Dtests.distribution=default -Druntime.java=8
```

**Reproduces locally?**:

**Applicable branches**:
7.x, 7.11 and 7.10

**Failure history**:

This test failed a lot [in the last 24 hours ](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'class:%22org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT%22'),sort:!(time,desc)))

**Failure excerpt**:

> "type" : "illegal_state_exception",
            "reason" : "extracted source isn't an object or an array",
            "stack_trace" : "[extracted source isn't an object or an array]; nested: IllegalStateException[extracted source isn't an object or an array];

**Comment by tlrx** [comment]
@jtibshirani this looks related to a change you made so I'm assigning this to you (but feel free to reassign if needed). I'm going to mute the test on 7.x, 7.11 and 7.10 branches.

**Comment by jtibshirani** [comment]
Sorry for the hassle @tlrx. I just had to adjust the skip version: we were testing against 7.10.1 but relevant bugfix is only available in 7.10.2 and later.
